pkg/bugreport/controlplane.go

```diff
@@ -21,26 +21,41 @@ func (c *Config) collectControlPlaneLogs() error {
 			),
 		})
 	if err != nil {
-		return fmt.Errorf("error getting control plane pods: %w", err)
+		c.completionFailure("Error getting control plane pods: %s", err)
+		goto afterPodLogCollection
 	}
 
 	for _, pod := range pods.Items {
 		cmd := []string{"kubectl", "logs", "-n", pod.Namespace, pod.Name}
 		outPath := path.Join(c.rootNamespaceDirPath(), pod.Namespace, rootPodDirName, pod.Name, commandsDirName, strings.Join(cmd, "_"))
 		if err := runCmdAndWriteToFile(cmd, outPath); err != nil {
-			return fmt.Errorf("error writing control pod logs: %w", err)
+			c.completionFailure("Error writing control pod logs: %s", err)
+			continue
 		}
 		c.completionSuccess("Collected report for Pod %s/%s", pod.Namespace, pod.Name)
 
 		if pod.Status.ContainerStatuses[0].RestartCount > 0 {
 			cmd = append(cmd, "--previous")
 			outPath = path.Join(c.rootNamespaceDirPath(), pod.Namespace, rootPodDirName, pod.Name, commandsDirName, strings.Join(cmd, "_"))
 			if err := runCmdAndWriteToFile(cmd, outPath); err != nil {
-				return fmt.Errorf("error writing previous control pod logs: %w", err)
+				c.completionFailure("Error writing previous control pod logs: %s", err)
+				continue
 			}
 			c.completionSuccess("Collected previous report for Pod %s/%s", pod.Namespace, pod.Name)
 		}
 	}
 
-	return nil
+afterPodLogCollection:
+	// Collect MeshConfig, secrets in control plane namespace
+	namespaceCmds := [][]string{
+		{"kubectl", "get", "meshconfig", "-n", c.ControlPlaneNamepace, "-A", "-o", "yaml"},
```

Comment:
Was passing both `-n` to scope to one namespace and `-A` to read from all namespaces intentional?

Reply:
It wasn't, fixed.